### PR TITLE
Visit super arguments even when no signature exists

### LINKED ISSUE #24551: Import incorrectly removed during transpilation
Currently I'm having an issue where I've imported a variable, and the import is being removed during transpilation.  The reference still remains, but the code breaks at runtime because the import is missing.

**TypeScript Version:**  3.0.0-dev.201xxxxx, 2.9.1, 2.9.0-dev.20180406

I was able to pinpoint the issue to `2.9.0-dev.20180406` but I am having issues creating a small repro case.

**Search Terms:** 
Import missing reference

**Code**
I'm having trouble creating a contained repro, but the issue is currently occurring when an imported reference is used only in super clause of a constructor.

**Expected behavior:**
The import for an imported reference should exist after transpilation.

**Actual behavior:**
The import is removed during transpillation.

**Playground Link:**  

**Related Issues:**

### PR DESCRIPTION
**Pull request by weswigham**
Fixes #24551

### CODE REVIEW

**Comment by mhegazy** [comment]
Please port to release-2.9 as well

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -18678,6 +18678,7 @@ namespace ts {
             if (node.expression.kind === SyntaxKind.SuperKeyword) {
                 const superType = checkSuperExpression(node.expression);
                 if (isTypeAny(superType)) {
+                    forEach(node.arguments, checkExpression); // Still visit arguments so they get marked for visibility, etc
```
Comment (mhegazy) [inline src/compiler/checker.ts]: @weswigham this should be `forEach(node.arguments, n=>{checkExpression(n);});` 

the return value from `checkExpression` is short-circuiting after the first argument.

Comment (mhegazy) [reply src/compiler/checker.ts]: Repro in https://github.com/Microsoft/TypeScript/issues/24729#issuecomment-395808993

Comment (weswigham) [reply src/compiler/checker.ts]: D'oh.

**Pull request merged**